### DISCUSSION TITLE
Some questions on the threshold setting/obtained spectra/calibration source

### DISCUSSION THREAD

**Discussion by Myrmi123** [OpenGammaProject/Open-Gamma-Detector#61] [Q&A] [answered]
Hi everyone,
recently I got my detector running with a Pico W, a 1"x1" Na(Tl) crystal and a single photomultiplier 🥳. Heres the background spectrum with my setting of the threshold:
![grafik](https://github.com/user-attachments/assets/0dd0c3cc-5d11-4bb1-8683-e598e6c784eb)

I have some questions now:
1. Is this setting "good"? The Instructions on Hackaday say: _"Turn it until you get to a certain point where the ACTivity LED goes from solid on (very bright) to blinking (more dimly), now just turn it a bit in the direction of less blinking to be sure to not get noise in your signal."_
But even turning it 4 full revolutions of the potentiometer down from the "permanent on point" didn't give me the gap in the lower end in the spectrum that is seen in the instructions and I got very high count rates at channels 1-4. The setting above is 19 full revolutions away from the pernanent on point.
Also this spectrum has no real "Peak" like the example spectras. On the lefthand side there is only a sharp cutoff and the peak is directrly right to it.

2. The highest count rates are still around channel 1-4. Is this intended or could this be an EMI issue? Currently I have no housing and ca. 5cm long cables between the two boards...

3. Are the counts around channel 3900 real or a measurement artifact? They are also present in the example spectrum (https://cdn.hackaday.io/images/4690701671310338015.png)

4. A more fundamental question: Is the background spectrum actually "real" radiation or is most of it due to dark counts of the sensor? Compared to the rather long intervalls that I know from geiger counters 10-50 clicks per second sounds a lot to me.

The next step now would be to calibrate the energy right? Do you have any suggestions for an easy to come by source for the calibration?
I have potassium (from food supplements) and some small uranium glass spheres at home, but I didn't notice any significant increase in cps when they are near the detector... Is radiation from such sources too weak? Could it work with some lead shielding to reduce the background? Or would something like thorinated welding rods be better?

Thanks in advance

Best regards,
Dennis

**Comment by NuclearPhoenixx** [comment]
Hi Dennis!

Glad you got it running, here are some answers to your questions:

1. Ideally it should not have this sharp cutoff. This could either be an EMI issue or there is still some tiny light leak to your SiPM. Try using it in a dark environment and see if that changes anything. However, it looks to me like you're already pretty close to how it should be according to the ADC bins in the bg spectrum. The spectrum and count rate also look pretty good. Only the cutoff looks optimizable.
2. That's weird, can you maybe post the settings of your device? Counts on channel 0 are most likely the quantitative cps correction for the ADC DNL issues of the original Pico. The rest shouldn't be there.
3. Those are just measurement artifacts from the ADC. Sometimes very high energy particles pass through the scintillator and you get peaks with a higher voltage than 3 Volts, which is the maximum supported by the electronics and ADC. Therefore, those higher counts get clipped to the end of the measurement range and bunch up like that, that's totally normal.
4. The background spectrum is indeed due to natural background radiation. It originates from all kind of natural isotopes around us, mostly from the natural decay chains, K-40 and a lot of scattered radiation. If you're more interested in this, the background spectrum follows the Landau distribution: https://en.wikipedia.org/wiki/Landau_distribution Dark counts of the sensor would be much lower in voltage and much higher in frequency. Scintillators are just that more sensitive than GMTs 😃 

Yes, the next step would be the energy calibration. Potassium is a good source, however you need a lot of it and measure quite a long time, because the detector sensitivity decreases a lot the higher energy you go. I would advice against any kind of uranium sources, because usually they have a lot of other decay products in them. These peaks will overlap and make it very difficult to assign them to their energy.

Maybe you can get your hands on some old vaccuum tubes with Cs-137 or ionizations smoke detectors with Am-241. You can also buy lutetium oxide to measure the natural isotope Lu-176. It's just a metal, but you can also grab yourself some LYSO scintillators (on ebay) that also contain the same isotope. Those come to my mind now.

Hope that helps! Cheers.

**Reply by Myrmi123** [reply]
Thanks for the quick reply :) 
Today I put the whole assembly in an empty coke can and closed the lid with some aluminum foil. This should have eliminated EMI and light related issues but didn't change anything.
I built a makeshift lead castle and encased the scintillator. This reduced the total count rate by a factor of ~10-20. Still, the high counts at channels 1-5 remained. However, they were attenuated by the same factor of 10-20.

Here are my user settings:
```C++
#define SCREEN_TYPE SCREEN_SSD1306  // Display type: Either SCREEN_SSD1306 or SCREEN_SH1106
#define SCREEN_WIDTH 128            // OLED display width, in pixels
#define SCREEN_HEIGHT 64            // OLED display height, in pixels
#define SCREEN_ADDRESS 0x3C         // See datasheet for Address; 0x3D for 128x64, 0x3C for 128x32
#define EVENT_BUFFER 50000          // Buffer this many events for Serial.print
#define TRNG_BITS 8                 // Number of bits for each random number, max 8
#define BASELINE_NUM 101            // Number of measurements taken to determine the DC baseline
#define CONFIG_FILE "/config.json"  // File to store the settings
#define DEBUG_FILE "/debug.json"    // File to store some misc debug info
#define DISPLAY_REFRESH 1000        // Milliseconds between display refreshs
#define BUZZER_FREQ 2700            // Resonance frequency of the buzzer
#define AUTOSAVE_TIME 900000        // 15 Minutes in ms, time between recording autosaves
// Display Types: Select only one! One of them must be true and the other one false
#define SCREEN_SH1106 false
#define SCREEN_SSD1306 true

struct Config {
  // These are the default settings that can also be changes via the serial commands
  // Do not touch the struct itself, but only the values of the variables!
  bool ser_output = true;         // Wheter data should be Serial.println'ed
  bool geiger_mode = false;       // Measure only cps, not energy
  bool print_spectrum = false;    // Print the finishes spectrum, not just chronological events
  size_t meas_avg = 2;            // Number of meas. averaged each event, higher=longer dead time
  bool enable_display = true;    // Enable I2C Display, see settings above
  bool enable_trng = false;       // Enable the True Random Number Generator
  bool subtract_baseline = true;  // Subtract the DC bias from each pulse
  bool cps_correction = false;    // Correct the cps for the DNL compensation
  bool enable_ticker = false;     // Enable the buzzer to be used as a ticker for pulses
  size_t tick_rate = 20;          // Buzzer ticks once every tick_rate pulses
```
I also changed the threshold setting just as the instruction says. Heres the background spectrum that I get:
![grafik](https://github.com/user-attachments/assets/bf4737b0-b9e9-45fb-a04d-805987cc1664)
I set the x-axis to log scaling. At channel 0 there is no count, The Landau distribution is not visible, count rates just increase towards lower channels and spike at channels 1-5
Do you have any further ideas, what could be the reason for this strange behaviour?
At work I have access to an oscilloscope and I could check the pulses next week.

**Reply by NuclearPhoenixx** [reply]
Hmm, okay. Maybe the coupling between the SiPM and scintillator is not ideal, what did you use there? And what bias voltage and PCB are you using for the SiPM? 

Please also share the active device settings via the serial command `read settings`. The code you posted is fine.

**Reply by Myrmi123** [reply]
I use the MicroFC SiPM Carrier Board with temperature compensation. Input voltage is 30.6V, the voltage between Ground and the cathode is then 29.55V. For the coupling I used this[ silicone grease](https://www.obi.de/p/6695217/liqui-moly-silicon-fett-transparent-100-g). 
These are the active device settings:
```
read settings
{
  "ser_output": true,
  "geiger_mode": false,
  "print_spectrum": false,
  "meas_avg": 2,
  "enable_display": true,
  "enable_trng": false,
  "subtract_baseline": true,
  "cps_correction": false,
  "enable_ticker": false,
  "tick_rate": 20
}
[#] Read settings file successfully.
```
I thought that maybe too much light escapes around the sensor, so I covered the rest of the polished scintillator surface with teflon tape. This also didn't change the behaviour.

**Reply by NuclearPhoenixx** [reply]
Thanks for the additional info. I have to say I'm running a bit out of ideas. It still looks to me like either a noise or a light problem. Either EMI and/or bad cables/connector on the electrical side or too much light losses or light leak on the optical side, both of which you have pretty much refuted.

Since you have disabled the "cps_correction" setting, it also unsettles me to see all of the counts at the very first channels. These shouldn't be there in the first place. In my mind this can only happen if the peak and hold circuit is read out just as it is being reset, which is most of the time a timing issue due to high count rates. This probably won't solve anything, but you can also try using one of the clean uf2 firmware files and then resetting the settings manually once.

Did you get a pre-assembled PCB or did you source and solder all the parts yourself?

**Reply by Myrmi123** [reply, marked as answer]
Installing the "clean firmware" did the trick, thanks 😄 The culprit is the LED on the Pi Pico W Board. With the stock firware it worked and this was the only change I did. Commenting out the LED in my version also lead to a correct spectrum. gpio_set_slew_rate doesn't work with the LED on the "W"-Board because the LED is controlled through the wireless chip. Apparently it is driven so hard that this causes some EMI interference. 
The spectrum now looks like this:
![grafik](https://github.com/user-attachments/assets/fe58649b-5a1e-4b56-a03a-c9a4dcc6970a)
My guess is that adding white PTFE-Tape to the surface of the crystall not covered by the sensor should move the peak of the distribution further right.
Only thing that looks strange is that the spectrum has a saw-like pattern:
![grafik](https://github.com/user-attachments/assets/f2ccf668-2eee-4df3-9fd8-19619d753a25)
The dips occur every 8 samples. Is this related to the ADC-Issue of the Pi Pico?

**Reply by NuclearPhoenixx** [reply]
Awesome! The oldest trick in IT just after "did you try turning it off and on again" 😂 
I did not know that about the Pico W, that's good to know!

I noticed the ADC pattern too and as far as I can tell it's "normal"... I'm pretty sure it's the ADC, but it doesn't affect the usability. You're probably going to use the SMA anyways, that'll get rid of most of that. Maybe the RP2350 will fix this as well!
